lib/aryn-sdk/aryn_sdk/partition/partition.py

```diff
@@ -0,0 +1,189 @@
+    return json.dumps(options)
+
+
+# Heavily adapted from lib/sycamore/data/table.py::Table.to_csv()
+def tables_to_pandas(data: dict) -> list[tuple[dict, Optional[pd.DataFrame]]]:
+    """
+    For every table element in the provided partitioning response, create a pandas
+    DataFrame representing the tabular data. Return a list containing all the elements,
+    with tables paired with their corresponding DataFrames.
+
+    Args:
+        data: a response from ``partition_file``
+
+    Example:
+         .. code-block:: python
+
+            from arynsdk.partition import partition_file, tables_to_pandas
+
+            with open("my-favorite-pdf.pdf", "rb") as f:
+                data = partition_file(
+                    f,
+                    aryn_api_key="MY-ARYN-TOKEN",
+                    use_ocr=True,
+                    extract_table_structure=True,
+                    extract_images=True
+                )
+            elts_and_dataframes = tables_to_pandas(data)
+
+    """
+    results = []
+    for e in data["elements"]:
+        if e["type"] == "table" and e["table"] is not None:
+            table = e["table"]
+            header_rows = sorted(
+                set(row_num for cell in table["cells"] for row_num in cell["rows"] if cell["is_header"])
+            )
+            i = -1
+            for i in range(len(header_rows)):
+                if header_rows[i] != i:
+                    break
+            max_header_prefix_row = i
+            grid_width = table["num_cols"]
+            grid_height = table["num_rows"]
+
+            grid = np.empty([grid_height, grid_width], dtype="object")
+            for cell in table["cells"]:
+                if cell["is_header"] and cell["rows"][0] <= max_header_prefix_row:
+                    for col in cell["cols"]:
+                        grid[cell["rows"][0], col] = cell["content"]
+                    for row in cell["rows"][1:]:
+                        for col in cell["cols"]:
+                            grid[row, col] = ""
+                else:
+                    grid[cell["rows"][0], cell["cols"][0]] = cell["content"]
+                    for col in cell["cols"][1:]:
+                        grid[cell["rows"][0], col] = ""
+                    for row in cell["rows"][1:]:
+                        for col in cell["cols"]:
+                            grid[row, col] = ""
+
+            header = grid[: max_header_prefix_row + 1, :]
+            flattened_header = []
+            for npcol in header.transpose():
+                flattened_header.append(" | ".join(OrderedDict.fromkeys((c for c in npcol if c != ""))))
+            df = pd.DataFrame(
+                grid[max_header_prefix_row + 1 :, :],
+                index=None,
+                columns=flattened_header if max_header_prefix_row >= 0 else None,
+            )
+            results.append((e, df))
+        else:
+            results.append((e, None))
+
+    return results
+
+
+def add_bbox_to_pdf():
```

Comment:
Is this a placeholder? Do you want to keep it the pypi release before it is implemented?

Reply:
dropped